src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/SyncTable.cs

```diff
@@ -77,11 +78,21 @@ internal static class SyncTable
         private static int s_freeEntryList;
 
         /// <summary>
-        /// The index of the lowest never used entry.  We skip the 0th entry and start with 1.
+        /// The index of the lowest never used entry.  We skip the 0th and 1st entries and start with 2.
         /// If all entries have been used, s_unusedEntryIndex == s_entries.Length.  This counter
         /// never decreases.
         /// </summary>
-        private static int s_unusedEntryIndex = 1;
+        private static int s_unusedEntryIndex = 2;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ref Entry UnsafeEntryRef(int i)
```

Comment:
I think it would be fine to use `UnsafeEntryRef` for `GetHashCode` too.

Reply:
Reverted one too many.